rfcs/proposed/dynamic_selection_customization/custom_backends.md

```diff
@@ -36,8 +36,9 @@ This proposal presents a flexible backend system based on a `backend_base` templ
 ### Key Components
 
 1. **`backend_base<ResourceType, Backend>`**: A proposed base class template that implements the core backend functionality using CRTP.
-2. **`default_backend<ResourceType, ResourceAdapter>`**: A proposed template that provides a complete backend implementation for any resource type, with optional adapter support.
-3. **SYCL specialization**: A proposed specialized implementation for `sycl::queue` resources that handles SYCL-specific event management and profiling.
+2. **`default_backend_impl<typename BaseResourceType, ResourceType, ResourceAdapter>`**: A proposed template that provides a complete backend implementation for any resource type, with optional adapter support.
+3. **`default_backend<ResourceType, ResourceAdapter>`**: A proposed template that determines the `BaseResourceType` from the `ResourceType` and `ResourceAdapter`. A developer creates a partial specialization of `default_backend` to create a specific backend.
+4. **A SYCL specialization of default_backend**: A specialized implementation for `sycl::queue` resources that handles SYCL-specific event management and profiling. Using an adapter, it is possible to reuse this for other types that can be adapted into a `sycl::queue`, such as a `sycl::queue *` or a struct that contains a `sycl::queue`.
```

Comment:
A SYCL specialization of default_backend_impl

Reply:
Ok, will change.